packages/web-components/fast-router/src/commands.ts

```diff
@@ -0,0 +1,198 @@
+import {
+    Mutable,
+    ElementStyles,
+    html,
+    HTMLView,
+    ViewTemplate,
+    FASTElementDefinition,
+    defaultExecutionContext,
+} from "@microsoft/fast-element";
+import { navigationParticipant } from "./participants";
+import { NavigationTransaction } from "./transaction";
+import { Router } from "./router";
+import { RouterConfiguration } from "./configuration";
+import { Transition } from "./transition";
+import {
+    Layout,
+    LayoutAndTransitionRouteDefinition,
+    HasElement,
+    HasTemplate,
+} from "./routes";
+import { Navigation } from "./navigation";
+
+export interface NavigationCommand {
+    execute(transaction: NavigationTransaction): Promise<void>;
+}
+
+export class Ignore implements NavigationCommand {
+    public async execute(transaction: NavigationTransaction): Promise<void> {}
+}
+
+export class Redirect implements NavigationCommand {
+    constructor(private redirect: string) {}
+
+    public async execute(transaction: NavigationTransaction): Promise<void> {
+        // TODO: generate new route based on redirect and route params
```

Comment:
What kind of example new route would be generated based on route params? Like if you specified a pattern like: `/foo/:id/:name` and your route params included something like `{ id: "bar", name: "bob" }` you'd want to generate `foo/bar/bob`?